README.md

```diff
@@ -59,43 +59,58 @@ dotnet new gcf-event
 ```
 
 That will create the same set of files as before, but the `Function`
-class now implements `ICloudEventFunction`. This is a function that
-responds to [CNCF Cloud Events](https://cloudevents.io/). The procedure
-for running a Cloud Event Function is exactly the same as for an
-HTTP Function.
-
-## Legacy Event Functions
-
-Google Cloud Functions support for events predates the CNCF Cloud
-Events initiative. "Legacy" event functions handle these events via
-the `ILegacyEventFunction<T>` type. The type parameter `T`
-represents the payload type of the event. Any class type that can be
-deserialized via `System.Text.Json` can be used, but the Functions
-Framework for .NET comes with the following payload types built-in,
-all within the `Google.Cloud.Functions.LegacyEvents` namespace:
-
-- `StorageObject` for Google Cloud Storage events
-- `FirestoreEvent` for Cloud Firestore events
-- `PubSubMessage` for Cloud Pub/Sub messages
-
-See the Google Cloud Functions ["Events and Triggers"
-documentation](https://cloud.google.com/functions/docs/concepts/events-triggers)
-for more information about the triggers available.
+class now implements `ICloudEventFunction<StorageObject>`. This is a

```

Comment:
This feels a little mixed now in that the line between "true" cloud events and converted legacy functions is blury. I believe it would be clearer if:

- The explanation focuses on (typed) Cloud Events with a note "there are no pure cloud events right now, will be adding more information here when there are".
- A ### title with something like Legacy events as Cloud Events, that starts with the paragraph marked as **Note:**, and includes the StorageObject example.
- The untyped Cloud Event section as it is now.

Reply:
The explanation is focused on the sample, which has to be for a concrete event type.
I actually *like* it being blurry between "true" cloud events and converted ones - because that should be regarded as an implementation detail, ideally. The note is really just in case anyone looks at the underlying HTTP request and gets confused.